src/main/scala/io/lenses/data/generator/domain/recursive/CustomerGenerator.scala

```diff
@@ -0,0 +1,74 @@
+package io.lenses.data.generator.domain.recursive
+
+import io.lenses.data.generator.config.DataGeneratorConfig
+import io.lenses.data.generator.domain.Generator
+import io.lenses.data.generator.kafka.Producers
+import org.apache.avro.SchemaBuilder
+import org.apache.avro.generic.GenericData
+import org.apache.avro.generic.GenericRecord
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.StringSerializer
+import org.joda.time.DateTime
+
+object CustomerGenerator extends Generator {
+  private val schema = SchemaBuilder
```

Comment:
We have avro4s exactly for this. Toavoid  manual schema building and rely on case classes.
Was there a specific reason to have it manual here?

Reply:
I might refactor this to use avro4s. Haven't touched this class far, so it should be the same code that we have in the master branch.